src/node/stage/approved.rs

```diff
@@ -2430,15 +2415,40 @@ impl Approved {
                 VoteTheirKnowledge { prefix, key_index } => {
                     self.cast_unordered_vote(core, Vote::TheirKnowledge { prefix, key_index })?
                 }
-                VoteSendNeighbourInfo { dst, nonce } => {
-                    self.cast_ordered_vote(AccumulatingEvent::SendNeighbourInfo { dst, nonce })
+                SendNeighbourInfo { dst, nonce } => {
+                    self.send_neighbour_info(core, dst, nonce, None)?
                 }
             }
         }
 
         Ok(())
     }
 
+    fn send_neighbour_info(
+        &mut self,
+        core: &mut Core,
+        dst: XorName,
+        nonce: MessageHash,
+        dst_key: Option<bls::PublicKey>,
+    ) -> Result<()> {
+        let proof_chain = self.shared_state.create_proof_chain_for_our_info(None);
```

Comment:
I think we should pass `shared_state.sections.knowledge_by_location(DstLocation::Section(dst))` to this function to make sure the attached proof chain would be trusted by the destination.

Reply:
You mean the proof_chain that passed to `Message::single_src` shall be `proof_chain = self.shared_state.sections.knowledge_by_location(DstLocation::Section(dst))`

Reply:
Like this:
```rust
let proof_chain = self.shared_state.create_proof_chain_for_our_info(Some(DstLocation::Section(dst));
```